cmd/dep/status.go

```diff
@@ -570,7 +570,12 @@ func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceMana
 							// upgrade, the first version we encounter that
 							// matches our constraint will be what we want.
 							if c.Constraint.Matches(v) {
-								bs.Latest = v.Revision()
+								// Latest should be of the same type as the Version.
+								if bs.Version.Type() == gps.IsSemver {
+									bs.Latest = v
+								} else {
```

Comment:
This doesn't account for plain versions - those will still show up with a rev. Can deal with it later, though.

Reply:
But we enter this version's updateability check block only for non-rev and non-plain version. LATEST for plain version would be empty. I think that's the right behavior.